src/open_dread_rando/files/custom_scenario.lua

```diff
@@ -230,6 +230,32 @@ function Scenario.UpdateProgressiveItemModels()
     Game.AddSF(0.1, "Scenario._UpdateProgressiveItemModels", "")
 end
 
+Scenario._BlastShieldTypes = {
+    doorshieldsupermissile = {
+        item = "ITEM_WEAPON_SUPER_MISSILE",
+        damage = {"SUPER_MISSILE", "ICE_MISSILE"},
+    },
+    door_shield_plasma = {
+        item = "ITEM_WEAPON_PLASMA_BEAM",
+        damage = {"PLASMA_BEAM"}
+    }
+}
+function Scenario._UpdateBlastShields()
+    for name, actordef in pairs(Game.GetEntities()) do
+        shield_type = Scenario._BlastShieldTypes[actordef]
+        if shield_type ~= nil and RandomizerPowerup.HasItem(shield_type.item) then
+            local shield = Game.GetActor(name)
+            for _, damage in ipairs(shield_type.damage) do
+                shield.LIFE:AddDamageSource(damage)
+            end
+        end
+    end
+end
+
+function Scenario.UpdateBlastShields()
+    Game.AddSF(0.1, "Scenario._UpdateBlastShields", "")
+end
+
```

Comment:
As I'm not familiar with damage sources:
Vanilla Plasma Shields are vulnerable against wave but wave is not dealing plasma damage? Cause otherwise you would destroy a plasma shield with wave beam even with these updates (except exactly that is patched out of a solo wave beam).

What happens if you play with the infamous non-progressive mode? Picking up wave beam first would mean you don't have Plasma in your inventory or does it? I know that picking up wave beam instantly gives you (wide+)plasma but I'm not sure about the inventory item of (wide+)plasma itself which you are checking here.

Reply:
it works correctly - you cannot open plasma doors with wave beam until you collect plasma beam

Reply:
Yeah, but still the last part is questionable for me: Playing with non-progressive, non-split beams and collecting wave beam first. If it doesn't give you plasma beam item for the inventory, plasma shields are not destroyable which would be wrong.

Reply:
There is no such thing as non-progressive, non-split beams. They're either progressive or split.

Reply:
Now, I'm confused.
If I currently generate a seed with rdv with non-progressive beams, you will have a wave beam item in the game which will behave like you have all beams when picked up.
So, you are saying you have completely erased this option and old rdvgame could technically be impossible with the new patcher because you only receive the single wave beam now?

Reply:
You understand correctly. As far as I'm concerned, the legacy non-progressive beam/missile options were never properly supported anyway. I'm not worried about breaking compat with them

Reply:
Uh, ok. Then it makes much more sense. I thought this will be a third option like progressive, split or "priority" mode where the best beam makes the other obsolete.
Then my case is not an issue at all.